js/test/Ice/idleTimeout/Client.ts

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+import { Ice } from "ice";
+import { Test } from "./Test.js";
+import { TestHelper } from "../../Common/TestHelper.js";
+
+const test = TestHelper.test;
+
+// The client and server have the same idle timeout (1s) and the server enables connection idle checks
+// (the default).
+// We verify that the server's idle check does not abort the connection as long as this connection receives
+// heartbeats, even when the heartbeats are not read off the connection in a timely manner.
+// To verify this situation, we use an OA with a 1-thread thread pool and use this unique thread for a long
+// synchronous dispatch (sleep).
+async function testIdleCheckDoesNotAbortConnectionWhenThreadPoolIsExhausted(
```

Comment:
This test checks the server-side behavior, and as I understand it, the server is the C++ server.

Whereas what we want to test here is the JS code.

Reply:
We cannot implement this test in JavaScript. If the event loop cannot process events it will not process heartbeat frames, but it won't be able to process timer events either.